### ISSUE TITLE
[Performance]: Hy3 MTP speculative decoding is slower than plain decode — HYV3MTP is missing `@support_torch_compile`

### ISSUE THREAD

**Issue by auroter** [vllm-project/vllm#48766]
## Proposal / observation

`HYV3MTP` in `vllm/model_executor/models/hy_v3_mtp.py` does not carry the
`@support_torch_compile` decorator. The target model runs compiled with CUDA
graphs, but every draft forward runs fully eager. The Hy3 draft layer is
itself a 192-expert MoE block plus a full-vocab logits GEMM, so one eager
draft cycle costs roughly a full graphed target step — which makes MTP
speculative decoding a net loss on this model family regardless of
acceptance rate.

`DeepSeekMTP` has carried the same decorator since #25109, with the same
forward shape (single decoder layer over `input_embeds` + `previous_hidden_states`).
`hy_v3_mtp.py` never received the equivalent change (byte-identical from
v0.24.0 through current main at the time of writing).

## Measurements

Hardware: 4x RTX PRO 6000 Blackwell (SM120), TP4, single-stream decode,
kv-cache fp8. Model: tencent/Hy3 (295B-A21B, 80 layers + 1 MTP layer).

Without the decorator (stock):

| config | tok/s | pos-0 acceptance |
|---|---|---|
| plain decode (W4A16 house quant) | 119 | — |
| MTP depth 1, v0.24.0 | 94 | 56.2% |
| MTP depth 1, v0.25.x | 90 | 57.6% |
| MTP depth 3, v0.24.0 | 85 | — |

Acceptance is in the expected band for the checkpoint, so the regression is
pure draft-cycle cost: speculation loses ~21-29% against plain decode.

With `@support_torch_compile` applied to `HYV3MTP` (two-line change, see
linked PR), on the official tencent/Hy3-FP8 checkpoint:

| config | tok/s | acceptance |
|---|---|---|
| plain decode | 96.9 | — |
| MTP depth 1 | 105.7 (+9.1% vs plain) | 57.7% pos-0 |
| MTP depth 2 | 87.2 | 44% pos-1 conditional |

Depth 1 flips from a net loss to a +9.1% gain. Depth 2 still loses, but for
checkpoint reasons rather than engine reasons: the Hy3 draft head is a
single layer trained for depth 1, and conditional acceptance collapses past
position 0.

## Fix

Add the decorator, matching DeepSeekMTP:

```python
from vllm.compilation.decorators import support_torch_compile

@support_torch_compile
class HYV3MTP(nn.Module):
    ...
```

## Disclosure

This issue was investigated and drafted with AI assistance (Claude); all
measurements were taken and verified on our hardware, and the text was
reviewed by the submitting engineer.
